packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -135,7 +135,7 @@ class NumberInput extends Component {
     /**
      * Specify how much the valus should increase/decrease upon clicking on up/down button
      */
-    step: PropTypes.number,
+    step: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
```

Comment:
```suggestion
    step: PropTypes.oneOf([PropTypes.number, 'any']),
```

not 100% sure if this works but could we narrow down "string" to "any"? or are there other string values we should be accounting for?

Reply:
I think we can do something like this:
```
PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf(['all'])
])
```
if it is preferred.  I will update my PR title.

Reply:
This is the only string value allowed for this attribute.

Reply:
I have made the suggested change to the proptype declaration. It will now throw a warning if any other string value is entered beside "any"

Reply:
thanks for the update! yeah if I'm not mistaken we can also change some of the type checks for strings to specifically look for "any" now